### PR TITLE
Fixing hybrid test target compilation issue

### PR DESCRIPTION
**Pull request by bhariharan**
and some cleanup

### CODE REVIEW

File: libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -63,12 +60,6 @@
 		8222729F19C39F4500FC537E /* XCTest.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 8222729E19C39F4500FC537E /* XCTest.framework */; };
 		822272A019C39F4500FC537E /* Foundation.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 82EFF2D316E7C57100A63CDF /* Foundation.framework */; };
 		822272A119C39F4500FC537E /* UIKit.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 82EFF30016E8049500A63CDF /* UIKit.framework */; };
-		822272C219C3A2A500FC537E /* CoreGraphics.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 82EFF30316E8049500A63CDF /* CoreGraphics.framework */; };
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj]: We were embedding a bunch of frameworks that aren't required.

File: libs/SalesforceHybridSDK/SalesforceHybridSDKTestApp/Plugins/SFTestRunnerPlugin.h [hunk 2]
```diff
@@ -22,9 +22,7 @@
  WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#import <Foundation/Foundation.h>
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDKTestApp/Plugins/SFTestRunnerPlugin.h]: These imports were unnecessary - they are transitively pulled in.

File: libs/SalesforceHybridSDK/SalesforceHybridSDKTestApp/Plugins/SFTestRunnerPlugin.m [hunk 3]
```diff
@@ -24,6 +24,7 @@
 
 #import "SFTestRunnerPlugin.h"
 #import "CDVPlugin+SFAdditions.h"
+#import <SalesforceHybridSDK/SalesforceHybridSDK.h>
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDKTestApp/Plugins/SFTestRunnerPlugin.m]: This was causing the compilation failure.

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

### Tests: 

Classname | Name|
--- | ---|
SmartSyncTestSuite | testSyncSObjectDetectConflictDelete|
SmartSyncTestSuite | testSyncSObjectDetectConflictUpdate|
SmartSyncTestSuite | testSyncSObjectRetrieve|
SyncManagerTests | testCleanResyncGhostsForMRUTarget|
SyncManagerTests | testCleanResyncGhostsForMRUTarget|
SyncManagerTests | testCleanResyncGhostsForMRUTarget|


## Current coverage for SalesforceAnalytics.framework is `67.16%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceHybridSDK.framework is `54.78%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceSDKCore.framework is `50.85%`
✅ *No files affecting coverage found*

---
## Current coverage for SmartStore.framework is `64.15%`
✅ *No files affecting coverage found*

---
## Current coverage for SmartSync.framework is `79.73%`
✅ *No files affecting coverage found*

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**